include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -242,6 +245,38 @@ struct __parallel_for_submitter<__internal::__optional_kernel_name<_Name...>>
     }
 };
 
+template <typename... _Name>
+struct __parallel_for_submitter</*_IsGPU=*/::std::true_type, __internal::__optional_kernel_name<_Name...>>
+{
+    // GPU specialization with strided access memory pattern
+    template <typename _ExecutionPolicy, typename _Fp, typename _Index, typename... _Ranges>
+    auto
+    operator()(_ExecutionPolicy&& __exec, _Fp __brick, _Index __count, _Ranges&&... __rngs) const
+    {
+        assert(oneapi::dpl::__ranges::__get_first_range_size(__rngs...) > 0);
+        _PRINT_INFO_IN_DEBUG_MODE(__exec);
```

Comment:
Should we move these debug statements into the calling `__parallel_for` to not have to duplicate them?

Reply:
Yes, I will do this.